server/lib/data/get-receipt-data.js

```diff
@@ -0,0 +1,32 @@
+const { internalStatuses, notifyStatuses } = require('./receipt-statuses')
+const { getReceipts } = require('../../lib/db')
+
+/**
+ * Gets the data to be displayed for sent messages from the receipts DB container.
+ *
+ * @param {string} messageId of the message to return stats on.
+ * @returns {object} stats of the sent message.
+ */
+module.exports = async messageId => {
+  const [failedToSend, notifyDelivered, notifyFailed, pendingSending, rateLimited, timeOfFirstSend, timeOfLastSend, toBeRetried] = await Promise.all([
```

Comment:
Would it be possible to move more logic to the server when querying by doing a `GROUP BY` status and `Count`?

Reply:
Good suggestion, I'll get this merged (as it does what I need right now) and take a look at it in detail tomorrow.